utils/scripts/ocm/ocm.py

```diff
@@ -1270,48 +1317,228 @@ def compare_with_old_version_file(self):
-        metavar="",
-        default="us-east-1",
     )
-    optional_create_cluster_parser.add_argument(
-        "--aws-instance-type",
+    aws_create_cluster_parser.add_argument(
+        "--aws-instance-type ",
         help="aws instance type",
         action="store",
         dest="aws_instance_type",
         metavar="",
         default="m5.2xlarge",
     )
-    optional_create_cluster_parser.add_argument(
-        "--num-compute-nodes",
+    aws_create_cluster_parser.add_argument(
+        "--num-compute-nodes ",
         help="Number of compute nodes",
         action="store",
         dest="num_compute_nodes",
         metavar="",
         default="3",
     )
-    optional_create_cluster_parser.add_argument(
-        "--openshift-version",
-        help="Openshift Version",
+
+    gcp_create_cluster_parser.add_argument(
+        "--gcp-sa-project-id ",
+        help="gcp service account project id",
+    )
+
+    gcp_create_cluster_parser.add_argument(
+        "--gcp-sa-priv-key-id ",
+        help="gcp service account private key id",
+    )
+
+    gcp_create_cluster_parser.add_argument(
+        "--gcp-sa-private-key ",
+        help="gcp service account private key",
+    )
+
+    gcp_create_cluster_parser.add_argument(
+        "--gcp-sa-client-id ",
+        help="gcp service account client id",
+    )
+
+    gcp_create_cluster_parser.add_argument(
+        "--gcp-sa-client-email ",
+        help="gcp service account client email",
+    )
+
+    gcp_create_cluster_parser.add_argument(
+        "--gcp-sa-client-cert-url ",
+        help="gcp service account client cert url",
+    )
+
+    gcp_create_cluster_parser.add_argument(
+        "--compute-nodes ",
+        help="Number of compute nodes",
         action="store",
-        dest="openshift_version",
+        dest="compute_nodes",
         metavar="",
-        default="",
+        default="2",
     )
-    optional_create_cluster_parser.add_argument(
-        "--channel-group",
-        help="Channel group name. Values can be stable or candidate.",
+
+    gcp_create_cluster_parser.add_argument(
+        "--region ",
+        help="gcp region",
         action="store",
-        dest="channel_group",
+        dest="region",
         metavar="",
-        default="",
+        default="us-central1",
     )
+    gcp_create_cluster_parser.add_argument(
+        "--compute-machine-type ",
+        help="compute machine type",
+        action="store",
+        dest="compute_machine_type",
+        metavar="",
+        default="custom-8-32768",
+    )
+    known_args = ""
+    try:
+        f = io.StringIO()
+        with redirect_stdout(f), redirect_stderr(f):
+            known_args = parser.parse_known_args()
+    except SystemExit:
+        pass
+    if known_args and "cloud_provider" in known_args[0]:
+        provider = known_args[0].cloud_provider
+        if provider == "aws":
+            required_create_cluster_parser.add_argument(
+                "--aws-account-id",
```

Comment:
wasn't it already defined at line 1324?

Reply:
Yes, one is exclusively for help content and another one is for getting input from user. This is kind of workaround in argparse that we have to do, if we are using if else stmt in between the user inputs.